### PR TITLE
Fix serialization of evaluation response.

### PR DESCRIPTION
**Pull request by przemekwitek**
Serialization and deserialization routines for `EvaluateDataFrameAction.Response` were not in sync since https://github.com/elastic/elasticsearch/pull/41937.
This PR fixes that.

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/EvaluateDataFrameActionResponseTests.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+public class EvaluateDataFrameActionResponseTests extends AbstractWireSerializingTestCase<Response> {
+
+    @Override
+    protected NamedWriteableRegistry getNamedWriteableRegistry() {
+        return new NamedWriteableRegistry(new MlEvaluationNamedXContentProvider().getNamedWriteables());
+    }
+
+    @Override
+    protected Response createTestInstance() {
+        String evaluationName = randomAlphaOfLength(10);
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/EvaluateDataFrameActionResponseTests.java]: Would be even better if we had a list of random size [0, 3], say, and try adding a couple of different metrics.

**Review by dimitris-athanasiou — COMMENTED** [review]
Good catch! Just left a suggestion to fortify a bit more.

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/EvaluateDataFrameActionResponseTests.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/EvaluateDataFrameActionResponseTests.java]: Done.

**Comment by przemekwitek** [comment]
> Good catch! Just left a suggestion to fortify a bit more.

Suggestion applied. PTAL

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Pull request merged**